aten/src/ATen/native/mps/operations/Distributions.mm

```diff
@@ -412,8 +412,8 @@ Tensor normal_mps(const Tensor& mean, const Tensor& std, c10::optional<Generator
   };
 
   return mps::random_mps_impl<int64_t>(result,
-                                       0.0,
-                                       1.0,
+                                       std::numeric_limits<int64_t>::min(),
+                                       std::numeric_limits<int64_t>::max(),
```

Comment:
Changed this upper bound from 1 to int64 max which solves the original problem by making the range wider. Used int64 max again b/c more numbers -> fewer collisions. I will run benchmarks tomorrow.